src/net/db.rs

```diff
@@ -1073,6 +1283,136 @@ impl PeerDB {
+        conn: &Connection,
+        used_slot: u32,
+    ) -> Result<Vec<ContractId>, db_error> {
+        let mut db_set = HashSet::new();
+        let qry = "SELECT smart_contract_id FROM stackerdb_peers WHERE peer_slot = ?1";
+        let dbs = query_rows(conn, qry, &[&used_slot])?;
+        for cid in dbs.into_iter() {
+            db_set.insert(cid);
+        }
+
+        Ok(db_set.into_iter().collect())
+    }
+
+    /// Get the slots for all peers that replicate a particular stacker DB
+    fn get_stacker_db_slots(
+        conn: &Connection,
+        smart_contract: &ContractId,
+    ) -> Result<Vec<u32>, db_error> {
+        let qry = "SELECT peer_slot FROM stackerdb_peers WHERE smart_contract_id = ?1";
+        let args: &[&dyn ToSql] = &[&smart_contract.to_string()];
+        query_rows(conn, qry, args)
+    }
+
+    /// Get a peer's advertized stacker DBs
+    fn static_get_peer_stacker_dbs(
+        conn: &Connection,
+        neighbor: &Neighbor,
+    ) -> Result<Vec<ContractId>, db_error> {
+        let used_slot_opt = PeerDB::find_peer_slot(
+            conn,
+            neighbor.addr.network_id,
+            &neighbor.addr.addrbytes,
+            neighbor.addr.port,
+        )?;
+        if let Some(used_slot) = used_slot_opt {
+            Self::get_stacker_dbs_by_slot(conn, used_slot)
+        } else {
+            Ok(vec![])
+        }
+    }
+
+    /// Get a peer's advertized stacker DBs by their IDs.
+    pub fn get_peer_stacker_dbs(&self, neighbor: &Neighbor) -> Result<Vec<ContractId>, db_error> {
+        PeerDB::static_get_peer_stacker_dbs(&self.conn, neighbor)
+    }
+
+    /// Update an existing peer's stacker DB IDs.
+    /// Calculates the delta between what's in the DB now, and what's in `dbs`, and deletes the
+    /// records absent from `dbs` and adds records not present in the DB.
+    /// Does nothing if the peer is not present.
+    pub fn update_peer_stacker_dbs<'a>(
+        tx: &Transaction<'a>,
+        neighbor: &Neighbor,
+        dbs: &[ContractId],
+    ) -> Result<(), db_error> {
+        let cur_slot_opt = PeerDB::find_peer_slot(
+            tx,
+            neighbor.addr.network_id,
+            &neighbor.addr.addrbytes,
+            neighbor.addr.port,
+        )?;
+        let cur_dbs_set: HashSet<_> = PeerDB::static_get_peer_stacker_dbs(tx, neighbor)?
+            .into_iter()
+            .collect();
+        let new_dbs_set: HashSet<ContractId> = dbs.iter().map(|cid| cid.clone()).collect();
+        let to_insert: Vec<_> = new_dbs_set.difference(&cur_dbs_set).collect();
+        let to_delete: Vec<_> = cur_dbs_set.difference(&new_dbs_set).collect();
+
+        let sql = "DELETE FROM stackerdb_peers WHERE smart_contract_id = ?1";
+        for cid in to_delete.into_iter() {
+            test_debug!("Delete Stacker DB for {:?}: {}", &neighbor.addr, &cid);
+            let args: &[&dyn ToSql] = &[&cid.to_string()];
+            tx.execute(sql, args).map_err(db_error::SqliteError)?;
+        }
```

Comment:
This deletes those stacker db entries for _all peers_, is that the intended behavior of this function?

Reply:
Nope, will fix